### PR TITLE
chore(rxPermission): decompose monolithic source

### PR DESCRIPTION
**Pull request by CITguy**
JIRA: https://jira.rax.io/browse/FRMW-361

### LGTMS
- [x] Dev LGTM
- [x] Dev LGTM

### CODE REVIEW

File: src/rxPermission/rxPermission.spec.js [hunk 1]
```diff
@@ -1,119 +0,0 @@
-/* jshint node: true */
-describe('rxPermission', function () {
-    describe('rxPermission Directive', function () {
```
Comment (CITguy) [inline src/rxPermission/rxPermission.spec.js]: moved to `src/rxPermission/scripts/rxPermission.spec.js`

File: src/rxPermission/rxPermission.spec.js [hunk 2]
```diff
@@ -1,119 +0,0 @@
...
-            expect(elFail.text().trim()).to.be.empty;
-        });
-
-        it('rxPermission: should not display text when user has none of multiple roles', function () {
-            expect(elFailRoles.text().trim()).to.be.empty;
-        });
-
-    });
-
-    describe('Permission', function () {
```
Comment (CITguy) [inline src/rxPermission/rxPermission.spec.js]: moved to `src/rxPermission/scripts/Permission.spec.js`

File: src/rxPermission/scripts/Permission.js [hunk 3]
```diff
@@ -102,32 +87,4 @@ angular.module('encore.ui.rxPermission', ['encore.ui.rxSession'])
...
- * @scope
- * @description
- * Simple directive which will show or hide content based on whether or not the user has the specified role. See
- * the `rxPermission` component {@link /encore-ui/#/components/rxPermission demo} for an example.
- *
- * @requires rxPermission.service:Permission
- *
- * @param {String} role - Name of required role.
- */
-.directive('rxPermission', function () {
```
Comment (CITguy) [inline src/rxPermission/scripts/Permission.js]: moved to `src/rxPermission/scripts/rxPermission.js`

**Comment by 100stacks** [comment]
LGTM

**Comment by bry4n** [comment]
LGTM

**Pull request merged**